here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/executors/write/UpdateFeature.kt

```diff
@@ -0,0 +1,206 @@
+        }
+        require(previousMetadata.nextVersion == null) {
+            "Previous metadata shouldn't have 'nextVersion' but it does (${previousMetadata.nextVersion})"
+        }
+
+        val tupleNumber = newFeatureTupleNumber(collection, feature.id, session)
+        val flags = resolveFlags(collection, session)
+        val newVersion = Version(previousMetadata.version.txn + 1)
+        val tuple = tuple(
+            session.storage,
+            tupleNumber,
+            feature,
+            metadataForNewVersion(previousMetadata, newVersion, feature, flags),
+            write.attachment,
+            flags
+        )
+
+        removeFeatureFromDel(collection, feature.id)
+        collection.history?.let { hstTable ->
+            insertPreviousVersionToHst(hstTable.name, cursorWithFeatureHead, newVersion)
+        }
+        cursorWithFeatureHead.close() // we don't need it anymore, TODO: refactor for nicer usage
+        updateFeatureInHead(collection, tuple, feature, newVersion, previousMetadata)
+        return tuple
+    }
+
+    private fun metaFromCursor(cursor: PgCursor): Metadata {
+        // TODO: maybe PgCursor.map()?
+        val txn: Int64 = cursor[PgColumn.txn]
+        return Metadata(
+            storeNumber = cursor[PgColumn.store_number],
+            updatedAt = cursor[PgColumn.updated_at],
+            createdAt = cursor[PgColumn.created_at],
+            authorTs = cursor[PgColumn.author_ts],
+            nextVersion = maybeVersion(cursor.column(PgColumn.txn_next)),
+            version = Version(txn),
+            prevVersion = maybeVersion(cursor.column(PgColumn.ptxn)),
+            uid = cursor[PgColumn.uid],
+            puid = cursor.column(PgColumn.puid) as? Int,
+            hash = cursor[PgColumn.hash],
+            changeCount = cursor[PgColumn.change_count],
+            geoGrid = cursor[PgColumn.geo_grid],
+            flags = cursor[PgColumn.flags],
+            id = cursor[PgColumn.id],
+            appId = cursor[PgColumn.app_id],
+            author = cursor.column(PgColumn.author) as? String,
+            type = cursor.column(PgColumn.type) as? String,
+            origin = cursor.column(PgColumn.origin) as? String
+        )
+    }
+
+    private fun maybeVersion(rawCursorValue: Any?): Version? {
+        return (rawCursorValue as? Int64)?.let {
+            if (it.toInt() == 0) {
+                null
+            } else {
+                Version(it)
+            }
+        }
+    }
+
+    private fun fetchFromHead(collection: PgCollection, featureId: String): PgCursor {
+        val quotedHeadName = quoteIdent(collection.head.name)
+        val quotedIdColumn = quoteIdent(PgColumn.id.name)
+        return session.usePgConnection().execute(
+            sql = """SELECT ${PgColumn.allColumns.joinToString(separator = ", ")}
+                     FROM $quotedHeadName
+                     WHERE $quotedIdColumn='$featureId'
```

Comment:
you can use $1 and pass featureId in args, then implementation should take care of all escaping